app/Http/Controllers/Admin/LoginController.php

```diff
@@ -82,7 +84,15 @@ public function login(LoginRequest $request)
                 $user = User::updateOrCreate($userCondition, $user);
                 # Set login for user
                 Auth::login($user, $request->filled('remember'));
-                return redirect("/admin");
+                # check redirect to
+                $checkURL = session()->get('url_previous');
+                if (!empty($checkURL) && $checkURL != route('login')) {
```

Comment:
you should use:
` if (empty($checkURL) || $checkURL == route('login')) {
       $checkURL="/"
}
return redirect($checkURL);
`